### PR TITLE
Small bug fixes

### PR DESCRIPTION
**Pull request by rccarper**
*Description of changes:*
* Enforcing min part size.
* Failing loudly for ContentMD5 headers.
* Not requiring Content-Type to be in a PutObject message for CreateMultipartUpload.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: source/s3_client.c [hunk 1]
```diff
@@ -180,6 +181,16 @@ struct aws_s3_client *aws_s3_client_new(
         return NULL;
     }
 
+    /* Cannot be less than zero.  If zero, use default. */
+    if (client_config->part_size != 0 && client_config->part_size < MB_TO_BYTES(s_s3_min_part_size_MB)) {
```
Comment (TingDaoK) [inline source/s3_client.c:185]: I tried download with small part_size, it was accpeted by the server. So, the min part size seems only applied to upload? Not sure, we should rule it out totally or not.

File: source/s3_client.c [hunk 2]
```diff
@@ -180,6 +181,16 @@ struct aws_s3_client *aws_s3_client_new(
...
+        AWS_LOGF_ERROR(
+            AWS_LS_S3_CLIENT,
+            "Cannot create client from client_config; part size is less than the minimum part size of %d MB.",
```
Comment (TingDaoK) [inline source/s3_client.c:188]: `part size is not zero(default) and less than the minimum part size of %d MB` Or something like that will be better

File: source/s3_client.c (hunk 1 above)
Comment (rccarper) [reply source/s3_client.c:185]: Yeah, I'm on the fence with that too.

File: source/s3_client.c (hunk 2 above)
Comment (rccarper) [reply source/s3_client.c:188]: It's subjective, but I think calling out the default case is redundant in the messaging.

**Review by TingDaoK — APPROVED** [review]

**Pull request closed without merging**